<a href="https://colab.research.google.com/github/mairaalejandrarueda1-ctrl/programacion1/blob/main/26_marzo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

QUIZZ MODULAR 26 MARZO

*Maira Alejandra Rueda Manrique*

In [ ]:
%%writefile pedirnotas.py
def pedir_nombre():
  return input("Ingrese el nombre del estudiante: ")
def pedir_notas():
  n1 = float(input("Ingrese la nota 1: "))
  n2 = float(input("Ingrese la nota 2: "))
  n3 = float(input("Ingrese la nota 3: "))
  n4 = float(input("Ingrese la nota 4: "))
  return n1, n2, n3, n4


Overwriting pedirnotas.py


In [ ]:
%%writefile promedio.py
import statistics

def calcular_promedio(n1, n2, n3, n4):
    return (n1 + n2 + n3 + n4) / 4

def mediana(n1, n2, n3, n4):
    return statistics.median([n1, n2, n3, n4])

def moda(n1, n2, n3, n4):
        return statistics.mode([n1, n2, n3, n4])


Overwriting promedio.py


In [ ]:
import importlib
import pedirnotas
importlib.reload(pedirnotas)
nombre = pedirnotas.pedir_nombre()

Ingrese el nombre del estudiante: Alejandra


In [ ]:
nombre

'Alejandra'

In [ ]:
notas = pedirnotas.pedir_notas()


Ingrese la nota 1: 3.8
Ingrese la nota 2: 5.0
Ingrese la nota 3: 2.4
Ingrese la nota 4: 4.0


In [ ]:
list(notas)

[3.8, 5.0, 2.4, 4.0]

In [ ]:
import promedio
import importlib

importlib.reload(promedio)
n1, n2, n3, n4 = notas
prom = promedio.calcular_promedio(n1, n2, n3, n4)
mediana = promedio.mediana(n1, n2, n3, n4)
moda = promedio.moda(n1, n2, n3, n4)

print("estudiante", nombre)
print("promedio", prom)
print("mediana", mediana)
print("moda", moda)

if prom >= 3:
  print("aprobado")
else:
  print("reprobado")


estudiante Alejandra
promedio 3.8000000000000003
mediana 3.9
moda 3.8
aprobado


### Análisis de las notas de Alejandra

El estudiante **Alejandra** obtuvo las siguientes calificaciones:

*   **Promedio:** 3.8
*   **Mediana:** 3.9
*   **Moda:** 3.8

**Interpretación:**

*   **Promedio (3.8):** Indica que el rendimiento general de Alejandra en sus cuatro notas es de 3.8. Este valor es superior al umbral de aprobación (3.0).
*   **Mediana (3.9):** Al ordenar las notas, el valor central es 3.9. Esto sugiere que al menos la mitad de sus notas son iguales o superiores a 3.9, lo cual es un buen indicador de consistencia.
*   **Moda (3.8):** La nota que más se repite es 3.8. Esto puede indicar que una o más de sus calificaciones fueron 3.8, o que esta fue la nota más frecuente.

**Estado:**

Dado que el promedio de Alejandra (3.8) es mayor o igual a 3.0, la estudiante **APROBÓ**.